assets/app/features/upgrade/upgrade.controller.js

```diff
@@ -3,53 +3,31 @@
 
     /**
      * @ngdoc function
-     * @name crowbarApp.controller:Upgrade7Controller
+     * @name crowbarApp.controller:UpgradeController
      * @description
-     * # Upgrade7Controller
+     * # UpgradeController
      * This is the controller that will be used across the upgrade process.
      */
     angular.module('crowbarApp.upgrade')
-        .controller('UpgradeController', Upgrade7Controller);
+        .controller('UpgradeController', UpgradeController);
 
-    Upgrade7Controller.$inject = ['$scope', '$translate', '$state', 'upgradeStepsFactory'];
+    UpgradeController.$inject = ['$scope', '$translate', '$state', 'upgradeStepsFactory'];
     // @ngInject
-    function Upgrade7Controller($scope, $translate, $state, upgradeStepsFactory) {
+    function UpgradeController($scope, $translate, $state, upgradeStepsFactory) {
         var vm = this;
         vm.steps = {
             list: [],
-            activeStep: {},
             nextStep: nextStep,
-            isLastStep: isLastStep
+            isLastStep: isLastStep,
+            stepFinished: upgradeStepsFactory.stepFinished
         };
 
         // Get Steps list from provider
-        vm.steps.list = upgradeStepsFactory.getAll();
-        refeshStepsList();
+        vm.steps.list = upgradeStepsFactory.steps;
```

Comment:
This actually comes from richa's PR, not this one. Stupid gituhb doent allow linked patches properly :/